Sources/SafeDICore/Models/Scope.swift

```diff
@@ -78,27 +78,31 @@ final class Scope: Hashable {
 
     func createScopeGenerator(
         for property: Property? = nil,
-        instantiableStack: OrderedSet<Instantiable> = [],
         propertyStack: OrderedSet<Property> = []
     ) throws -> ScopeGenerator {
-        if let cycleIndex = instantiableStack.firstIndex(of: instantiable) {
```

Comment:
this was _way_ too aggressive a check before. We care about regenerating a property within a stack, not regenerating an instantiable (which may or may not be a constant, and may or may not have different property names)